core/src/main/java/com/graphhopper/storage/GraphHopperStorage.java

```diff
@@ -128,6 +140,32 @@ public int getEdges() {
         return (T) cg;
     }
 
+    public CHGraphImpl getCoreGraph(Weighting weighting) {
+        if (chGraphs.isEmpty())
+            throw new IllegalStateException("Cannot find graph implementation");
+        Iterator<CHGraphImpl> iterator = chGraphs.iterator();
+        while(iterator.hasNext()){
+            CHGraphImpl cg = iterator.next();
+            boolean isWeighting = cg.getWeighting().getName() == weighting.getName();
+            isWeighting = cg.getWeighting().getFlagEncoder().toString() == weighting.getFlagEncoder().toString();
+            //TODO This is very ugly. The weightings are the same but different instances so just getWeighting != weighting :(
```

Comment:
Can we fix this now before committing to the main branch?

Reply:
At the minute, the second line makes the first one obsolete as it will always be overwritten.

Reply:
And I actually don't see `isWeighting` getting used anywhere